modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2375,15 +2376,129 @@ def _dict_func(self, func, axis, *args, **kwargs):
         if "axis" not in kwargs:
             kwargs["axis"] = axis
 
-        def dict_apply_builder(df, func_dict={}):
-            # Sometimes `apply` can return a `Series`, but we require that internally
-            # all objects are `DataFrame`s.
-            return pandas.DataFrame(df.apply(func_dict, *args, **kwargs))
+        # Hopefully, this is a full list of non-reduction DataFrame methods that are
+        # available in `apply()`, otherwise labels precomputation would work incorrect:
+        non_reduction_methods = (
```

Comment:
feels a bit clumsy, but allows to precompute result labels if the list is filled correctly